### PR TITLE
Add CI test to ensure libtinfo isn't in binary

### LINKED ISSUE #4253: v0.14.7 depends on libtinfo again
## Description

In https://github.com/purescript/purescript/pull/4069, we dropped the dependency on libtinfo in order to avoid constant compatibility issues across different linux distributions and versions. In v0.14.7, that dependency seems to have inadvertently reappeared.

## To Reproduce

Download the latest `linux64` binary bundle from the releases page and run the following commands:

```
$ tar xzf linux64.tar.gz
$ cd purescript/
$ ldd purs | grep libtinfo
	libtinfo.so.5 => /lib/x86_64-linux-gnu/libtinfo.so.5 (0x00007efe968ab000)
```

## Expected behavior

The `ldd purs` command should not mention `libtinfo` in its output.

## PureScript version

0.14.7

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Fixes #4253. WIP.

I've only added a CI test based on Harry's comment. Also, when I ran `stack build --haddock --fast`, a `lib/purescript-ast.cabal` file was generated. Any idea why that happened? Running `stack purge` and then the same command doesn't generate the file again.

---

**Checklist:**

- [ ] Added a file to CHANGELOG.d for this PR (see CHANGELOG.d/README.md)
- [ ] Added myself to CONTRIBUTORS.md (if this is my first contribution)
- [x] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by JordanMartinez** [comment]
Can I get an approval on this? Then we can make another alpha release and unblock the Spago PR.

**Review by thomashoneyman — APPROVED** [review]

**Pull request merged**
